File: EduChemSuite.API/EduChemSuite.API/Models/BaseModel.cs
using System.ComponentModel.DataAnnotations;

namespace EduChemSuite.API.Models;

public abstract class BaseModel
{
    public Guid? Id { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public Boolean? IsActive { get; set; }
}

File: EduChemSuite.UI/UI/src/app/_models/BaseModel.ts
export abstract class BaseModel {
  id?: string; // Use string for GUID in TypeScript
  createdAt?: Date;
  updatedAt?: Date;
  isActive?: boolean;

  constructor(
    id?: string,
    createdAt?: Date,
    updatedAt?: Date,
    isActive?: boolean
  ) {
    this.id = id;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.isActive = isActive;
  }
}
